### ISSUE TITLE
Uncaught TypeError: Cannot read property 'x' of undefined

### ISSUE THREAD

**Issue by zepumph** [phetsims/friction#48]
```
FrictionKeyboardDragHandler.js?bust=1511819598427:34 Uncaught TypeError: Cannot read property 'x' of undefined
    at FrictionKeyboardDragHandler.onDrag (FrictionKeyboardDragHandler.js?bust=1511819598427:34)
    at FrictionKeyboardDragHandler.step (KeyboardDragHandler.js?bust=1511819598427:216)
    at Magnifier.step (Magnifier.js?bust=1511819598427:264)
    at FrictionView.step (FrictionView.js?bust=1511819598427:85)
    at Sim.stepSimulation (Sim.js?bust=1511819598427:831)
    at Sim.stepOneFrame (Sim.js?bust=1511819598427:764)
    at Sim.runAnimationLoop (Sim.js?bust=1511819598427:747)
```

This happens when a11y is not enabled, with fuzzMouse,

**Comment by zepumph** [comment]
Tagging @jessegreenberg so he is aware.

**Comment by zepumph** [comment]
I looked into this. Sometimes it takes a while of fuzzing (like 1 minute) to get this error. When it is paused in the stack I can't get any good info about why the old value is undefined. I tried two things to mitigate to no avail:
1. Add a check after each time I set `oldValue` to debugger if it is undefined (never gets called)
2. Copy all of the Vector2s just in case it was a reference problem (I guess not).

@phet-steele is there any way that you could help me try to figure out the cause of this?

**Comment by jessegreenberg** [comment]
@phet-steele helped determine that this was a precision issue. In KeyboardDragHandler, we were calling `onDrag` erroneously because there were cases where `this.positionProperty.get()` was out of drag bounds by ~0.000000001. After constraining bounds, `newPosition` and `positionProperty.value` were not equal, so were calling onDrag.

This is fixed by preventing behavior in KeyboardDragHandler so that no changes are made if no keys are down, which is a good check to add. But there might still be a problem in that FrictionModel.move allows position to go ~0.000000001 outside of bounds.

**Comment by zepumph** [comment]
Do you recommend adding a `.erode( .00000001)`?

**Comment by jessegreenberg** [comment]
After the above commit we can't reproduce the issue. @zepumph could you please review? I am not sure if any more work is worthwhile in FrictionModel.move. But something about that function allows movement to go slightly out of these bounds in FrictionKeyboardDragHandler:
```js
dragBounds: new Bounds2(
        -FrictionModel.MAX_X_DISPLACEMENT, // left bound
        FrictionModel.MIN_Y_POSITION, // top bound
        FrictionModel.MAX_X_DISPLACEMENT, // right bound
        2000 ) // bottom bound, arbitrary because the model stops the motion when the top atoms collide with the bottom book
    } );
```

**Comment by zepumph** [comment]
I can't seem to recreate it. Thanks! 

While working on this, I was wondering if there is potential weirdness about references vs clones of Vector2. We treat is as a value type, but it is still an object. I am wondering if the following lines couldn't be updated in `FrictionKeyboardDragHandler.js`

```js
      startDrag: function() {
        oldValue = model.positionProperty.get();
      },
      onDrag: function() {
        var newValue = model.positionProperty.get(); // --> .get().copy();
        model.move( { x: newValue.x - oldValue.x, y: newValue.y - oldValue.y } );

        // update the oldValue for the next onDrag
        oldValue = model.positionProperty.get(); // --> .get().copy();
      },
```
@jessegreenberg is it surprising to you that we aren't getting weird reference bugs here?

**Comment by jessegreenberg** [comment]
That does seem more robust @zepumph, good idea. It seems like we would run into reference bugs if `model.positionProperty` changes between `startDrag` and `onDrag`.

**Comment by zepumph** [comment]
Thanks @jessegreenberg, I agree. Closing.
